src/bin/ion/commands/generate/mod.rs

```diff
@@ -154,3 +163,23 @@ impl IonCliCommand for GenerateCommand {
         Ok(())
     }
 }
+
+impl GenerateCommand {
+    // Prints warning messages for Java code generation
+    fn print_java_code_gen_warnings() {
+        println!("{}","WARNING: Code generation in Java does not support any `$NOMINAL_ION_TYPES` data type.(For more information: https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#built-in-types)".yellow().bold());
+        println!(
+            "{}",
+            "Optional fields in generated code are represented with the wrapper class of that primitive data type and are set to `null` when missing."
+                .yellow()
+                .bold()
+        );
+        println!("{}", "When the `writeTo` method is used on an optional field and if the field value is set as null then it would skip serializing that field.".yellow().bold());
+    }
+
+    // Prints warning messages for Rust code generation
+    fn print_rust_code_gen_warnings() {
+        println!("{}","WARNING: Code generation in Rust does not support any `$NOMINAL_ION_TYPES` data type.(For more information: https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#built-in-types)".yellow().bold());
+        println!("{}","Code generation in Rust does not support optional/required fields. It does not have any checks added for this on read or write methods.".yellow().bold());
```

Comment:
For the warnings about point-in-time limitations, let's say so.

```suggestion
        println!("{}","Code generation in Rust does not yet support optional/required fields. It does not have any checks added for this on read or write methods.".yellow().bold());
```

And if there's a GitHub issue we could link, so much the better.